src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Cache.Memory;
+using Microsoft.Framework.FileSystemGlobbing;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// Utility methods for <see cref="ITagHelper"/>'s that support attributes containing file globbing patterns.
+    /// </summary>
+    public class GlobbingUrlBuilder
+    {
+        private static readonly char[] PatternSeparator = new[] { ',' };
+
+        private readonly FileProviderGlobbingDirectory _baseGlobbingDirectory;
+
+        public GlobbingUrlBuilder(IMemoryCache cache, IFileProvider fileProvider, PathString requestPathBase)
+        {
+            Cache = cache;
+            FileProvider = fileProvider;
+            RequestPathBase = requestPathBase;
+            _baseGlobbingDirectory = new FileProviderGlobbingDirectory(fileProvider, fileInfo: null, parent: null);
```

Comment:
Note: `fileProvider` could be `null` with how it is right now (not sure if the `FileProviderGlobbingDirectory` is ok with that).